selfdrive/ui/qt/widgets/map.cc

```diff
@@ -0,0 +1,500 @@
+#include <cmath>
+
+#include <QDebug>
+#include <QJsonDocument>
+#include <QJsonObject>
+
+#include "selfdrive/common/util.h"
+#include "selfdrive/common/swaglog.h"
+#include "selfdrive/common/params.h"
+#include "selfdrive/ui/qt/widgets/map_helpers.h"
+#include "selfdrive/ui/qt/widgets/map.h"
+
+
+const int PAN_TIMEOUT = 100;
+const bool DRAW_MODEL_PATH = false;
+const qreal REROUTE_DISTANCE = 25;
+const float MAX_ZOOM = 17;
+const float MIN_ZOOM = 14;
+const float MAX_PITCH = 50;
+const float MIN_PITCH = 0;
+const float MAP_SCALE = 2;
+
+static void clearLayout(QLayout* layout) {
```

Comment:
already defined in selfdrive/ui/qt/offroad/networking.cc

Reply:
should we make a qt_common library? In this case I thought it was better to copy it than to link against some code from the networking widget.

Reply:
https://github.com/commaai/openpilot/blob/master/selfdrive/ui/qt/util.h